subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/ApiGroovyCompiler.java

```diff
@@ -297,6 +306,17 @@ private void copyJavaCompilerResult(ApiCompilerResult javaCompilerResult) {
         }
     }
 
+    private static boolean isFatalException(org.codehaus.groovy.control.CompilationFailedException e) {
+        if (e instanceof org.codehaus.groovy.control.MultipleCompilationErrorsException) {
+            for (Message message : ((MultipleCompilationErrorsException) e).getErrorCollector().getErrors()) {
+                if (message instanceof ExceptionMessage) {
```

Comment:
To me it is not immediately obvious why this makes it fatal, a comment would be nice.

Reply:
Maybe include an example exception stacktrace as well?